feature/match screen result

NadeenXo opened the pull request:
match result screen
<img width="410" height="832" alt="image" src="https://github.com/user-attachments/assets/f4623b98-d053-4455-a5d1-557ed54a872b" />

NadeenXo committed:
min screen

NadeenXo committed:
item match list

NadeenXo committed:
text

NadeenXo committed:
margin

NadeenXo committed:
todo

NadeenXo committed:
start of the pager

NadeenXo committed:
start of the pager

NadeenXo committed:
deleted unused file

NadeenXo committed:
add the fragment to nav graph

NadeenXo committed:
Merge branch 'develop' into feature/match-screen-ui

# Conflicts:
#	app/src/main/res/values/strings.xml

baraa0abd reviewed (commented):
How can I navigate or view the screen?

<img width="493" height="638" alt="Image" src="https://github.com/user-attachments/assets/97fd11e6-ff79-4e2d-9d42-118b34473939" />

NadeenXo committed:
add navigation to match fragment

NadeenXo commented:
> How can I navigate or view the screen?
> 
> <img alt="Image" width="493" height="638" src="https://private-user-images.githubusercontent.com/168970211/489914820-97fd11e6-ff79-4e2d-9d42-118b34473939.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTgwMzQ1NzUsIm5iZiI6MTc1ODAzNDI3NSwicGF0aCI6Ii8xNjg5NzAyMTEvNDg5OTE0ODIwLTk3ZmQxMWU2LWZmNzktNGUyZC05ZDQyLTExOGIzNDQ3MzkzOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwOTE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDkxNlQxNDUxMTVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00NzE3OGQzYjQxZmZkMzQ4N2M1NTc2MzkxNDVlOTQyNjIxNjNiY2I1YzFiYjdlZTg5MzE4MWE2YzY1YzlmNmMzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.mDeyAGMqsUEAzrs6tkLiMX_8xQVxCdC3GeCgVD0tw-c">

check now @baraa0abd

baraa0abd reviewed (changes requested):
WELL DONE 👏👏, JUST SMALL CHANGES IF THAT IS POSSIBLE 


figma:

<img width="353" height="161" alt="Image" src="https://github.com/user-attachments/assets/4f45434f-23d5-4b61-afea-9a0dab8219bb" />

design:

<img width="325" height="157" alt="Image" src="https://github.com/user-attachments/assets/eec35eb6-9a54-485d-9a12-891b6a0e3b44" />

------------------------------------------------------------------------------------------------
1. more padding than needed between the two icons and at the start of the " movio " word

<img width="325" height="157" alt="Image" src="https://github.com/user-attachments/assets/b0007a8d-46b3-4070-90b1-c454e2d42327" />

NadeenXo committed:
fix margins

NadeenXo committed:
Merge branch 'develop' into feature/match-screen-ui

# Conflicts:
#	app/src/main/java/com/karrar/movieapp/ui/main/MainActivity.kt
#	app/src/main/res/values/strings.xml

NadeenXo committed:
merged with dev

baraa0abd committed:
Refactor MatchResultsFragment UI and variable names

This commit refactors the UI of the `MatchResultsFragment` and renames some variables for clarity.

Specifically, the following changes were made:
* In `MatchResultsFragment.kt`:
    * Renamed `item` to `itemMatch` in `OnPageChangeCallback` and when setting initial item.
* In `fragment_match_result.xml`:
    * Renamed data variable `item` to `itemMatch`.
    * Added new data variables `itemMovie` and `listener`.
    * Replaced the existing movie details card with a new layout that includes play trailer and save buttons, and uses the new data variables.
    * Updated the "View Details" button's `onClick` to use `itemMatch`.

baraa0abd committed:
Merge branch 'feature/match-screen' into feature/match-screen-ui

# Conflicts:
#	app/src/main/java/com/karrar/movieapp/ui/match/MatchItemUI.kt
